pool_automation/roles/aws_manage/library/test_stateful_set.py

```diff
@@ -93,21 +103,60 @@ def test_find_ubuntu_image(ec2):
     assert 'UNSUPPORTED' not in image.description
 
 
-def test_create_instances(ec2):
+def test_AwsEC2Launcher(ec2):
+    launcher = AwsEC2Launcher()
     params = PARAMS._replace(role='test_create')
-    instances = create_instances(ec2, params, 2)
+    instances = launcher.launch(params, 2, ec2=ec2)
 
     assert len(instances) == 2
+    assert len(launcher._ids) > 0
```

Comment:
Accessing implementation details often leads to fragile tests, is it really worth here?

Reply:
Why do think so? It would lead to collection of waiters just to serve instance launching across regions. Waiter class can encapsulate that well itself.